### ISSUE TITLE
config deepMerge issue

### ISSUE THREAD

**Issue by beckyi** [imzbf/md-editor-rt#179]
### Describe the issue

Could you please check the highlight instance merge?

I updated `v.4.20.0` from `v.4.13.3` (version of md-editor-rt)

I had previously set up instances.  (cropper / highlight / screenfull)

```tsx
//example
import {
  MdEditor,
  config
} from 'md-editor-rt'
import cropper from 'cropperjs'
import highlight from 'highlight.js'
import screenfull from 'screenfull'

config({
  editorExtensions: {
    highlight: {
      instance: highlight, // <-- error occurred
    },
    screenfull: {
      instance: screenfull,
    },
    cropper: {
      instance: cropper,
    },
  },
)

const MarkdownEditor = ({markdown}: {markdown: string}) => {
     ...
     return (
             <MDEditor
               ref={editorRef}
               modelValue={markdown}
               noMermaid={true}
               noKatex={true}
               noIconfont={true}
               noPrettier={true}
        />
     )
}
```

```js
// package.json
"highlight.js": "^11.10.0",
```

But type error occurred because of highlight's readonly property as shown in the photo below.

<img width="1017" alt="image" src="https://github.com/user-attachments/assets/6ca6faf6-6578-4fa5-af1f-75210d3a5a0d">

cc. deepMerge source added in [v.4.14.0](https://github.com/imzbf/md-editor-rt/commit/a81dafac50de326438dacbde218d0ffeba40ec83)

### Procedure version

browser

**Comment by imzbf** [comment]
Following your example, I'm unable to reproduce this issue. Could you provide a complete project?

**Comment by beckyi** [comment]
I think the ts config settings are different.

```
//tsconfig.json
{
  "compilerOptions": {
    "target": "ESNext",
    "lib": ["ES2019", "DOM"],
    "allowJs": true,
    "alwaysStrict": true, //<-
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "module": "ESNext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "removeComments": true,
    "noImplicitAny": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "compileOnSave": true,
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules"]
}
```

I reproduced from the feature/ui branch of [my practice project](https://github.com/JJStudyRoom/Decalcomanie/tree/feature/ui).

```zsh
$ git clone --branch feature/ui git@github.com:JJStudyRoom/Decalcomanie.git

#after install
$ cd ./packages/jaen-cafe && yarn dev

# http://localhost:3001/md
```

p.s. If the error is not reproduced, open the 'tsconfig.json' window, press 'ctrl + s'(save), and then build again.

<img width="1210" alt="image" src="https://github.com/user-attachments/assets/024c1ea2-93bc-4f7f-aed3-6b37975e6c5e">

**Comment by imzbf** [comment]
fixed in 4.20.1, have a try.
